# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp
# Commit: f18e61be455d
# Full Hash: f18e61be455d493f10a2cc651cd254859894e02e
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-02-26 03:52:35
# Regressor Bug: 1521879
# File Overlap Count: 1
# Description:
#   Bug 1521879 - Part 1: IPC-based MediaTransport implementation r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D17273
# ==============================================================================

diff -r 8136a17fbec9 -r f18e61be455d media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp
--- a/media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp	Mon Feb 25 21:48:20 2019 +0000
+++ b/media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp	Mon Feb 25 21:50:42 2019 +0000
@@ -702,7 +702,7 @@
   }
 }
 
-void MediaPipeline::SendPacket(MediaPacket& packet) {
+void MediaPipeline::SendPacket(MediaPacket&& packet) {
   ASSERT_ON_THREAD(mStsThread);
   MOZ_ASSERT(mRtpState == TransportLayer::TS_OPEN);
   MOZ_ASSERT(!mTransportId.empty());
@@ -1284,7 +1284,7 @@
           ("%s sending %s packet", mPipeline->mDescription.c_str(),
            (isRtp ? "RTP" : "RTCP")));
 
-  mPipeline->SendPacket(packet);
+  mPipeline->SendPacket(std::move(packet));
 }
 
 nsresult MediaPipeline::PipelineTransport::SendRtcpPacket(const uint8_t* aData,